daemon/algod/api/server/v2/handlers.go

```diff
@@ -0,0 +1,447 @@
+			return internalError(ctx, err, errFailedToParseTransaction, v2.Log)
+		}
+
+		response.Txn = encoded
+
+		if txn.ConfirmedRound != 0 {
+			r := uint64(txn.ConfirmedRound)
+			response.ConfirmedRound = &r
+
+			response.ClosingAmount = &txn.ApplyData.ClosingAmount.Raw
+			response.SenderRewards = &txn.ApplyData.SenderRewards.Raw
+			response.ReceiverRewards = &txn.ApplyData.ReceiverRewards.Raw
+			response.CloseRewards = &txn.ApplyData.CloseRewards.Raw
+
+			response.AssetIndex = computeAssetIndexFromTxn(txn, v2.Node.Ledger())
+		}
+
+		return ctx.JSON(http.StatusOK, response)
+	}
+
+	// We didn't find it, return a failure
+	err := errors.New(errTransactionNotFound)
+	return notFound(ctx, err, err.Error(), v2.Log)
+}
+
+// getPendingTransactions is a generalized version of the get pending transactions endpoints for code reuse.
+func (v2 *Handlers) getPendingTransactions(ctx echo.Context, max *uint64, format *string, addrFilter *string) error {
+	var addrPtr *basics.Address
+
+	if addrFilter != nil {
+		addr, err := basics.UnmarshalChecksumAddress(*addrFilter)
+		if err != nil {
+			return badRequest(ctx, err, errFailedToParseAddress, v2.Log)
+		}
+		addrPtr = &addr
+	}
+
+	handle, err := getCodecHandle(format)
+	if err != nil {
+		return badRequest(ctx, err, errFailedParsingFormatOption, v2.Log)
+	}
+
+	txns, err := v2.Node.GetPendingTxnsFromPool()
+	if err != nil {
+		return internalError(ctx, err, errFailedLookingUpTransactionPool, v2.Log)
+	}
+
+	// TODO: What should I put in here? MatchAddress uses this to check the FeeSink so I think this is fine.
+	spec := transactions.SpecialAddresses{
+		FeeSink:     basics.Address{},
+		RewardsPool: basics.Address{},
+	}
+
+	// Convert transactions to msgp / json strings
+	encodedTxns := make([]string, 0)
+	for _, txn := range txns {
+		// break out if we've reached the max number of transactions
+		if max != nil && uint64(len(encodedTxns)) >= *max {
+			break
+		}
+
+		// continue if we have an address filter and the address doesn't match the transaction.
+		if addrPtr != nil && !txn.Txn.MatchAddress(*addrPtr, spec) {
+			continue
+		}
+
+		// Encode the transaction and added to the results
+		encodedTxn, err := encode(handle, txn)
+		if err != nil {
+			return internalError(ctx, err, errFailedLookingUpTransactionPool, v2.Log)
+		}
+		encodedTxns = append(encodedTxns, encodedTxn)
+	}
+
+	return ctx.JSON(http.StatusOK, generated.PendingTransactionsResponse{
+		TopTransactions:   encodedTxns,
+		TotalTransactions: uint64(len(txns)),
+	})
+}
+
+// GetPendingTransactions gets a list of unconfirmed transactions currently in the transaction pool.
+// (GET /v2/transactions/pending)
+func (v2 *Handlers) GetPendingTransactions(ctx echo.Context, params generated.GetPendingTransactionsParams) error {
+	return v2.getPendingTransactions(ctx, params.Max, params.Format, nil)
+}
+
+// GetPendingTransactionsByAddress gets a list of unconfirmed transactions currently in the transaction pool by address.
```

Comment:
GetPendingTransactionsByAddress takes an Algorand address and returns its associated list of unconfirmed transactions currently in the transaction pool.